### PR TITLE
Eliminate Quasar's dependency on mimir

### PR DESCRIPTION
**Pull request by alissapajer**
[ch3725]

### CODE REVIEW

**Comment by alissapajer** [comment]
Snapshot released as `91.2.5-c14d648`.

File: mimir/src/main/scala/quasar/mimir/evaluate/QuasarImpl.scala [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        "UUID",
+        StoreKey.stringIso composePrism stringUuidP),
+      "TableRef",
+      rValueTableRefP(rValueSqlQueryP))
+
+  val pTableStore = MimirPTableStore[F](precog, PreparationLocation)
+
+  val pushdown = Ref.of[F, Pushdown](Pushdown.EnablePushdown).map(p => new PushdownControl(p))
+
+  val federation = MimirQueryFederation[Fix, F](precog, pushdown)
```
[the remark this answers is not in the document]
Comment (alissapajer) [reply mimir/src/main/scala/quasar/mimir/evaluate/QuasarImpl.scala]: Changed pushed :+1:

File: mimir/src/main/scala/quasar/mimir/evaluate/QuasarImpl.scala [hunk 2]
```diff
@@ -76,7 +77,7 @@ final class QuasarImpl[F[_]: ConcurrentEffect: Monad: MonadQuasarErr: PhaseResul
 
   val pTableStore = MimirPTableStore[F](precog, PreparationLocation)
 
-  val pushdown = Ref.of[F, Pushdown](Pushdown.EnablePushdown).map(p => new PushdownControl(p))
+  val pushdown = new PushdownControl(Ref.unsafe[F, Pushdown](Pushdown.EnablePushdown))
```
Comment (wemrysi) [inline mimir/src/main/scala/quasar/mimir/evaluate/QuasarImpl.scala]: If we're going to use `unsafe` here, then we should wrap the `new QuasarImpl` in an `F.delay` over in `QuasarImpl.apply`. Alternatively the `QuasarImpl` class could accept a `PushdownControl` argument and we construct the `Ref` in `QuasarImpl.apply`.

Comment (djspiewak) [reply mimir/src/main/scala/quasar/mimir/evaluate/QuasarImpl.scala]: Is there a way we can just not use `Ref.unsafe`? It's meant for some rather weird situations that are better handled with the 2-effect constructors.

Comment (alissapajer) [reply mimir/src/main/scala/quasar/mimir/evaluate/QuasarImpl.scala]: Yeah, we can.

**Pull request merged**